src/i_sdlsound.c

```diff
@@ -878,13 +943,40 @@ static int I_SDL_StartSound(sfxinfo_t *sfxinfo, int channel, int vol, int sep)
         return -1;
     }
 
-    snd = sfxinfo->driver_data;
+    snd = GetAllocatedSoundBySfxInfoAndPitch(sfxinfo, pitch);
```

Comment:
Seems weird that we do this whole lookup by pitch even when pitch shifting isn't turned on (ie. snd_pitchshift=false). The logic makes sense, right?